java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,250 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+public class RangeOptions {
+    public interface ScoreRange {
+        String toArgs();
+    }
+
+    public interface LexRange {
+        String toArgs();
+    }
+
+    public interface IRangeQuery {
+        String getStart();
+
+        String getEnd();
+
+        Limit getLimit();
+    }
+
+    public interface IScoredRangeQuery extends IRangeQuery {}
```

Comment:
lets move everything do with with ScoredRangeQuery to another file: say `ScoredRangeOptions`.

Reply:
Rename to `ScoredRangeQuery`.  No `I` needed.